### PR TITLE
Add information about RFC process

### PR DESCRIPTION
**Pull request by zuiderkwast**

### CODE REVIEW

**Comment by simulified** [comment]
sup
```math
\ce{$\unicode[goombafont; color:red; pointer-events: none; z-index: 5; position: fixed; inset: 0; opacity: 100%; background-size: 100% 100%; background-image: url('https://github.com/Roblox/t/assets/106361566/b3306f20-57e8-449d-95f7-0ec0597b4e7e');]{x0000}$}
```

File: README.md
Comment (madolson) [inline README.md:4]:
```suggestion
This repository is a collection of feature proposals and descriptions of changes to Valkey
```

File: README.md
Comment (madolson) [inline README.md:5]:
```suggestion
that require some more detail than just the text in a pull request or an issue.
```

File: README.md
Comment (madolson) [inline README.md:10]:
```suggestion
the pull request number that initially added an RFC to refer to the change. For example,
```

File: README.md
Comment (madolson) [inline README.md:17]: I think I would also include there is a standard for quality of an RFC on the content. If something is very vague and unclear, it's probably worth improving it before merging it. We won't reject it based off the proposal though.

File: README.md
Comment (madolson) [inline README.md:21]: I think we should decide that now. "Draft", "Published", "Approved", and "Rejected".

File: README.md
Comment (madolson) [inline README.md:31]: Why not make it strict?

File: README.md
Comment (madolson) [inline README.md:40]:
```suggestion
* Links to related material such as issues, pull requests, papers, or other references.
```

File: README.md
Comment (madolson) [inline README.md:39]:
```suggestion
* Design considerations. A description of the design constraints and requirements
  for the proposal.
* Specification. A more detailed description of the feature including 
  why certain design decisions have been made. Comparisons with
  similar features in other projects.
```
A slightly different wording to emphasize the constraints and incorporate the decision making inside the specification.

**Review by madolson — COMMENTED** [review]
Overall looks pretty good to me, some minor suggestions and wording improvements.

File: README.md
Comment (zuiderkwast) [reply README.md:21]: Yes. So let's define what these mean and choose terms that are unambigous.

- Draft = non yet merged PR (like this one). I don't think there's any reason to actually spell this out in the document itself. It's enough that the PR is not yet merged.
- Published = RFC merged to this repo but no decision made. PEPs seem to use Draft for this and they accept changes in this stage. I think Published can be interpreted as if the feature has been published, but it's just the RFC. Can we find a better term for this? I propose "Proposed". This is the stage where most of the discussion happens, as in Request For Comments. Perhaps we don't need to spell out a status at all. An RFC without a status is implicitly in this status.
- Approved = Specification approved by core team, awaiting implementation. Minor changes can still be requested in the actual implementation PR. (PEPs use "Accepted" for this.)
- Rejected = Not going to be implemented.

The PEP process has one more status "Final", meaning the feature has been implemented, merged, released. We should update an RFC to mention in which Valkey version it was released and add links to the implementation PR.

File: README.md
Comment (zuiderkwast) [reply README.md:31]: We want it to have less friction, be less formal than PEPs and RFCs right?

I think we can require this content but we don't have to enforce a specific order and headings within the document IMO. If somebody already has a document (like RDMA.md, TRIGGERS.md, CLUSTER.md) it can be added without too much rewrite.

I'll rewrite this part.

File: README.md
Comment (zuiderkwast) [reply README.md:39]: Why not keep the specification concise and separate out the "why"?

Small comparisons may make sense to incorporate in the specification, but larger ones may be better to move to its own section or subsection.

**Comment by zuiderkwast** [comment]
Updated. @madolson PTAL.

File: README.md
Comment (madolson) [reply README.md]: I'm ok with what you ended up.

File: README.md
Comment (madolson) [reply README.md]: I guess I would just like to have a strict set of stuff you "must" have, but give a lot of flexibility in those sections.

File: README.md
Comment (madolson) [reply README.md]: Should we add the list of states and what they mean here? Right now it's unclear what informational is unless you go look it up in the official IETF documentation.

File: README.md
Comment (zuiderkwast) [reply README.md]: Please suggest a formulation.

File: README.md
Comment (zuiderkwast) [reply README.md]: I ended up just adding Proposed, Approved and Rejected. I imagined they can't be misunderstood, but they probably can. I'll define them.

File: README.md
Comment (madolson) [inline README.md:37]:
```suggestion
The Valkey RFC format is not a strict format, but should include the following section unless they are unnecessary for the proposal you are submitting.
```

Just to say, you should use this unless you have a reason not to use this.

**Review by madolson — APPROVED** [review]
LGTM

**Comment by zuiderkwast** [comment]
I can't merge. I don't have permissions.

**Comment by madolson** [comment]
Should be fixed now.

File: README.md
Comment (madolson) [inline README.md:1]: Can you comment about this.

**Pull request merged**
